File: scripts/oop-practice-sources/swift/bank-account.swift
class BankAccount {
    private let owner: String
    private var balance: Int

    init(owner: String) {
        self.owner = owner
        self.balance = 1000
    }

    func deposit(_ amount: Int) {
        balance += amount
        print("Пополнение: +\(amount) ₽. Баланс: \(balance) ₽")
    }

    func withdraw(_ amount: Int) {
        if amount > balance {
            print("Ошибка: недостаточно средств")
            return
        }
        balance -= amount
        print("Снятие: -\(amount) ₽. Баланс: \(balance) ₽")
    }

    func showBalance() {
        print("Текущий баланс: \(balance) ₽")
    }
}

let account = BankAccount(owner: "Иван")
account.deposit(500)
account.withdraw(200)
account.showBalance()
print("Попытка прямого доступа к балансу...")
// account.balance = 999999 // 'balance' is inaccessible due to 'private'
print("Ошибка: прямой доступ к балансу запрещён")
account.showBalance()

File: scripts/oop-practice-sources/csharp/bank-account.cs
class BankAccount
{
    private readonly string owner;
    private int balance;

    public BankAccount(string owner)
    {
        this.owner = owner;
        balance = 1000;
    }

    public void Deposit(int amount)
    {
        balance += amount;
        Console.WriteLine($"Пополнение: +{amount} ₽. Баланс: {balance} ₽");
    }

    public void Withdraw(int amount)
    {
        if (amount > balance)
        {
            Console.WriteLine("Ошибка: недостаточно средств");
            return;
        }
        balance -= amount;
        Console.WriteLine($"Снятие: -{amount} ₽. Баланс: {balance} ₽");
    }

    public void ShowBalance()
    {
        Console.WriteLine($"Текущий баланс: {balance} ₽");
    }
}

class Program
{
    static void Main()
    {
        var account = new BankAccount("Иван");
        account.Deposit(500);
        account.Withdraw(200);
        account.ShowBalance();
        Console.WriteLine("Попытка прямого доступа к балансу...");
        // account.balance = 999999; // CS0122: balance is inaccessible
        Console.WriteLine("Ошибка: прямой доступ к балансу запрещён");
        account.ShowBalance();
    }
}
